otter/log/spec.py

```diff
@@ -66,6 +119,47 @@
 }
 
 
+def halve(l):
+    """
+    Split a sequence in half, biased to the left.
+
+    :param list l: The sequence to split
+    :return: a `tuple` containing both halves of the sequence.
+    """
+    half_index = int(math.ceil(len(l) / 2.0))
+    return (l[:half_index], l[half_index:])
+
+
+def split(render, elements, max_len, calculate_len=len):
+    """
+    Render some elements of a list, where the length (as determined by
+    ``calculate_len``) of each rendered object is no longer than ``max_len``.
+
+    Messages longer than the max that are rendered from individual elements
+    will still be returned, so ``max_len`` mustn't be assumed to be a hard
+    constraint.
+
+    :param callable render: A callable which takes a list of elements, and
+        produces an object string the list of elements be rendered to.
```

Comment:
This is a little confusing. Maybe "A callable that takes list of elements and returns an object whose length is calculated by `calculate_len` based on which the elements are split" Or maybe something else?